superset-frontend/src/SqlLab/components/SqlEditor.jsx

```diff
@@ -242,6 +245,13 @@ class SqlEditor extends React.PureComponent {
     }
   }
 
+  onBeforeUnload(event) {
+    if (this.props.latestQuery?.state === 'running') {
+      event.preventDefault();
+      this.stopQuery();
```

Comment:
One more question about this: can we make sure that we stop only queries when the DB is in sync mode? if the DB is in async mode the expectation I don't think we should ever cancel the query by navigating away from the browser.

Reply:
can we make it at least configurable? I’m happy to go for the extra mile if needed and add a new extra param to the db config if no objections. We really need to cancel all db queries both in sync and async modes.

This is because we’d like to go with async mode to utilise the celery workers but we still expect a lot of accidental cross joins, missing WHERE clauses and unwanted long running queries. In this case people will just close the browser and that’s making other people impossible running new queries. DBs like snowflake and redshift start queuing queries very quickly and easily could block using the db for long hours. Or DBAs have to kill unwanted queries manually.

Furthermore if it works only in sync mode then the stop button is basically doing nothing for non hive/presto databases.

Let me know your thoughts 🙇

Reply:
Sure, if there's a legit use case let's make it configurable. I like the idea of having a per-DB option, this way for Hive we could have it off (since queries can take hours), and for Postgres (say) we could have it on.

@yousoph, any thoughts here?

> Furthermore if it works only in sync mode then the stop button is basically doing nothing in async mode with non hive/presto databases.

Oh, sorry if I wasn't clear, I meant stopping a query by **navigating away**. If we press the stop button we should still cancel async queries, regardless of the configuration flag.

Reply:
This is now configurable at `Database -> Performance -> Cancel query on window unload event` and defaults to False to keep the current behaviour. Commit: e5223ecb03f3e48f2c9d0b43a851ed9eefb74ff6